client/src/pages/Drafts/DraftDetails.js

```diff
@@ -0,0 +1,108 @@
+import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+
+import { ResourceService, MemberService, useService } from "../../services";
+
+const style = {
+	secondary: "#bc2a2a",
+};
+
+const initialUserDetails = {
+	email: "UNK Email",
+	github_id: "UNK github ID",
+	id: "UNK ID",
+	is_admin: "UNK",
+	name: "NA Name",
+};
```

Comment:
you are right about using null/undefined here, I used an initial data and left it like this because I had an issue with the testing here, let's see if it makes sense to you. In the context of separation of concerns I think it's a good practice to extract side effects from components, in this case, the useEffect that makes the service call, by putting this useEffect under a custom hook outside the component, and update the state when service call resolves,  it's very easy to inject this service and mock the behaviour, this allows us to do isolated component testing of the component, otherwise we are bound to do the integration test. I don't know if I explained myself well :D

Reply:
I'd generally think about it like:

```mermaid
flowchart
    page[Page]
    service[Service]
    page <--> service
    api[(API/data)]
    service <-.-> api
    comp1[Component]
    comp2[Component]
    page --> comp1
    page --> comp2
```

"Components" are purer (although you can have some that take callbacks for sending data up to the parent), you can test those in isolation. The "pages" are responsible for coordinating with the services, so you test those in more of an integration way.

Reply:
we can talk about this if and when possible :) , I find it beneficial to mock the service calls at the page ( as you described them) level.